### DISCUSSION TITLE
PyPrimeMesh CAD *.scdoc issue?

### DISCUSSION THREAD

**Discussion by dkowollik** [ansys/pyprimemesh#873] [Q&A] [answered]
Hi,

if I try the following lines of code, I get a "CAD import failed." error using the *.scdoc file. 

```python
from ansys.meshing import prime
prime_client = prime.launch_prime()
model = prime_client.model
file_io = prime.FileIO(model)
import_params = prime.ImportCadParams(model=model)
file_io.import_cad(
        file_name='some_model.scdoc',
        params=import_params
    )
prime_client.exit()
```
Traceback:
```python
  File "C:\Python310_64_ansys\lib\site-packages\ansys\meshing\prime\core\fileio.py", line 586, in import_cad
    import_result = super().import_cad(temp_file_name, params)
  File "C:\Python310_64_ansys\lib\site-packages\ansys\meshing\prime\autogen\fileio.py", line 692, in import_cad
    result = self._comm.serve(self._model, command_name, self._object_id, args=args)
  File "C:\Python310_64_ansys\lib\site-packages\ansys\meshing\prime\internals\error_handling.py", line 577, in wrapper_error_code
    raise PrimeRuntimeError(
ansys.meshing.prime.internals.error_handling.PrimeRuntimeError: CAD import failed.
```

The default seems to be: `cad_reader_route=prime.CadReaderRoute.PROGRAMCONTROLLED`. If I use `WORKBENCH`, the import works fine: 

```python
from ansys.meshing import prime
prime_client = prime.launch_prime()
model = prime_client.model
file_io = prime.FileIO(model)
import_params = prime.ImportCadParams(model=model, cad_reader_route=prime.CadReaderRoute.WORKBENCH)
file_io.import_cad(
        file_name='some_model.scdoc',
        params=import_params
    )
prime_client.exit()
```

Which reader is used if the default option PROGRAMCONTROLLED is use? 
I thought that the importer loops by default through all available reader routes in order to find an appropriate one.

**Comment by waltersma** [comment]
Hi @dkowollik

For SCDOC files with PROGRAMCONTROLLED the SPACECLAIM reader would be used.   For a summary, see [here](https://prime.docs.pyansys.com/version/stable/user_guide/fileio.html#cad-reader-routes).  

Is it only a particular file you mention that fails?

**Reply by dkowollik** [reply]
Hi @waltersma,
for me this happens to any SCDOC file.

**Reply by aglowinski-danfoss** [reply]
Is there any update to this topic?

**Reply by waltersma** [reply]
Hi @dkowollik, @aglowinski-danfoss, just to confirm there is a change to the behaviour here for program controlled where since 0.7.0 and Ansys 25R1 SCDOC is now read using Discovery rather than SpaceClaim.  You must have the applications installed for this to work and licenses available.  Please can you confirm this is the case?

**Comment by dkowollik** [comment, marked as answer]
Now, the following code works fine for me. I tested this with 0.8.0 and Ansys 25R2.
```
from ansys.meshing import prime
prime_client = prime.launch_prime()
model = prime_client.model
file_io = prime.FileIO(model)
import_params = prime.ImportCadParams(model=model)
file_io.import_cad(
        file_name='some_model.scdoc',
        params=import_params
    )
prime_client.exit()
```

**Reply by waltersma** [reply]
Thanks for providing the feedback @dkowollik !

**Reply by waltersma** [reply]
I understand that to be 0.8.0 with Ansys 25R1?

**Reply by dkowollik** [reply]
Yes, it was 25R1 with SP2.
